Vhar 7146 automatic kulut valikatselmukseen

rckopteri opened the pull request.

rckopteri committed:
Lisää urakka_paatos tauluun kulu_id

Tällä varmistetaan kulun poisto, jos päätös poistetaan

rckopteri committed:
Lisää kulu_id jeesql asetuksiin

rckopteri committed:
Tallenna kulu oikealle päätöstyypille

ja poista se, jos päätös poistetaan

rckopteri committed:
Siirrä välikatselmuksen päätöksen koodeja erilliseen tiedostoon

Helpottaa kuluihin ja lupausbonuksiin/-sanktioihin liittyvien koodien hallintaa

rckopteri committed:
Päivitä kuluihin ja päätöksiin liittyvät sql haut

rckopteri committed:
Kierrä kulun käsittelyn käyttöoikeudet

mahdollistamalla käyttöoikeuden tarkistus irrallaan kulun luonnista. Tämä mahdollistaa välikatselmuksessa kulun luonnin.

rckopteri committed:
Estä käyttöliittymästä

hoitovuoden päätkseen liittyvien kulujen lisääminen, mutta mahdollista niiden näyttäminen, jos ne luotu välikatselmuksen kautta.

rckopteri committed:
Varmista yksikkötestein, että kulun luominen

toimii välikatselmuksen päätöksen luomisessa ja poistossa

rckopteri committed:
Poista cljs virhe,

joka aiheutui, kun ajan-muokkaus ei ollut ui:lta sallittua

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1042__.sql

rckopteri committed:
Korjaa ajan-muokkaus funktio

rckopteri committed:
Pidennä huomattavasti timeouttia

rckopteri committed:
Tavoitehinnan 3% yksikkötesti

vielä kulun lisäämisnäkökulmalla

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

rckopteri committed:
Päivitä kulujen cypress testit

jpluomajoki-solita commented on src/clj/harja/palvelin/palvelut/kulut/valikatselmukset.clj:
```diff
@@ -369,125 +295,56 @@
+          urakka (first (q-urakat/hae-urakka db urakka-id))
+          hoitokauden-alkuvuosi (::valikatselmus/hoitokauden-alkuvuosi tiedot)
+          ;; Rakennetaan valittu hoitokausi
+          #_#_valittu-hoitokausi [(pvm/hoitokauden-alkupvm hoitokauden-alkuvuosi)
```
Kommentoidun koodin voisi poistaa

jpluomajoki-solita commented on src/cljs/harja/views/urakka/kulut/kulut.cljs:
```diff
@@ -589,7 +590,7 @@
                                                       (= (:tehtavaryhma (last kohdistukset))
                                                         (:id (avain->tehtavaryhma tehtavaryhmat avain))))
-                                  :disabled (not= 0 haetaan)
+                                  :disabled true
```
Pitäisikö tämä radiobutton poistaa kokonaan?

rckopteri replied:
Mielestäni ei. Se on disabled, jotta sitä ei käytetä, mutta kun päätöksen yhteydessä on tällainen kulu luotu, niin se pystytään näyttämään samalla käyttöliittymällä.

jpluomajoki-solita committed:
Korjaa lupausbonuksen poistaminen

jpluomajoki-solita reviewed (approved):
* Kulua ei poisteta, kun tehdään oikaisu.
  * Sama pätee myös lupausbonuksiin, siihen näköjään jäänyt bugi. Tein commitin tähän pullariin, joka korjaa sen.
* Tavoitehinnan ylityksen ja alituksen kulut ovat molemmat miinusmerkkisiä. Pitää varmistaa kummin päin ne menee ja korjata.

jpluomajoki-solita commented:
`Urakoitsija kirjaa kulun harjaan`-teksti pitäisi myös poistaa. Pitäisikö johonkin myös laittaa tieto siitä, että kulut luodaan automaattisesti, eikä niitä tarvi enää tehdä itse?

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

rckopteri committed:
Merkitse tavoitehinnan alituksesta tuleva kulu tilaajan maksettavaksi

rckopteri committed:
Varmista testillä, että tavoitehinnan alituksen kulu on positiivinen

rckopteri committed:
Poista kulut, jos tehdään tavoitehinnan oikaisu

rckopteri committed:
Varmista yksikkötestillä

että tavoitehinnan oikaisu poistaa myös mahdollisesti päätöksen tekohetkellä syntyneen kulun.

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

rckopteri committed:
Laitetaan ehdolle vihjeteksti automaattisuudesta

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1043__.sql

rckopteri committed:
Lisää tarkistus, että kattohinnan

ylityksessä ei kulua luoda, jos urakoitsijan maksua ei ole asetettu tai se on nolla. Tämä tulee, kun koko ylityksen maksuosuus siirretään seuraavalle vuodelle pienentämään budjettia.

rckopteri committed:
Korjaa kulun summa

rckopteri committed:
Refaktoroi vanhaa funktiota - poista do

rckopteri committed:
Korjaa yksikkötestit

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	test/clj/harja/palvelin/palvelut/tuck_remoting/ilmoitukset_test.clj
#	tietokanta/src/main/resources/db/migration/V1_1044__.sql

rckopteri committed:
Muuta päätöksestä syntyvän kulun lisätietotekstit

rckopteri committed:
Muuta tavoitehinnan alituksesta syntyvän kulun

merkki negatiivisesta positiiviseksi, koska se on tilaajan maksettava kulu eli ikäänkuin bonus urakoitsijalle.

rckopteri committed:
Kevyt refaktorointi

rckopteri committed:
Piilota Hoitovuoden päätös radiobutton

jos hoitovuoden päätöstä ei ole valittuna, eli se on automatic kulu päätöksestä.

rckopteri committed:
Tavoitehinnan ylityspäätös poistuu gc2:sella

Lisää logia ongelman selvittämiseksi

rckopteri committed:
Kokeillaan vielä gc2:sta varten

rckopteri committed:
Kokeile alustaa lomakkeet ennen hakuja

rckopteri committed:
Poista debug lokituksia

rckopteri committed:
Lisätään yksikkötesti

jossa vielä varmistetaan, että tavoitehinnan alitus aiheuttaa max 3% bonuksen

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	src/cljc/harja/pvm.cljc

rckopteri committed:
Korjaa välikatselmuksen summa,

kun tavoitehinnan oikaisuja on tehty usealle eri vuodelle.

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1045__.sql

jpluomajoki-solita commented on src/clj/harja/kyselyt/valikatselmus.sql:
```diff
@@ -6,7 +6,7 @@ FROM urakka_tavoite ut
                     FROM tavoitehinnan_oikaisu t
                     WHERE NOT t.poistettu
-                    GROUP BY t."urakka-id", t."hoitokauden-alkuvuosi") t ON (ut.urakka = t."urakka-id")
+                    GROUP BY t."urakka-id", t."hoitokauden-alkuvuosi") t ON (ut.urakka = t."urakka-id" AND t."hoitokauden-alkuvuosi" = 2022)
```
Onko tämä nyt ihan oikein?

rckopteri committed:
Vaihda kovakoodattu vuosi muuttujaksi

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1046__.sql

rckopteri committed:
Refaktorointia

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

# Conflicts:
#	tietokanta/src/main/resources/db/migration/V1_1048__.sql

rckopteri committed:
Lisää muutkin hoitovuoden päättämiseen

liittyvät tehtäväryhmät paaria -porukkaan, joita ei näytetä käyttöliittymällä ellei ole pakko.

rckopteri committed:
Merge branch 'develop' into VHAR-7146-automatic-kulut-valikatselmukseen

rckopteri committed:
Kokeile onko timeoutin muuttamisella

vaikutuksia yksikkötestien toimivuudelle ci putkessa
